src/extension.ts

```diff
@@ -171,41 +171,121 @@ class RunLineInEditor {
+            while (command.trim().endsWith("`"))
+            {
+                // concatenate all lines into a single command
+                lineNumber ++;
+                command = command.replace("`", "") + this.StripComments(source.document.lineAt(lineNumber).text);
+            }
+            return command;
+        } 
+        else
+        {
+            // execute only the selected text
+            const selectionStart = source.selection.start;
+            const selectionEnd = source.selection.end;
+            if (selectionStart.line === selectionEnd.line)
+            {
+                return this.StripComments(source.document.getText(new Range(selectionStart, selectionEnd)));
+            }
+            else
+            {
+                command = this.StripComments(source.document.lineAt(selectionStart.line).text.substring(selectionStart.character));
+                for (let index = selectionStart.line+1; index <= selectionEnd.line; index++) {
+                    var line = this.StripComments(source.document.lineAt(index).text);
+                    if (line.startsWith("az"))
+                    {
+                        window.showErrorMessage<any>("Multiple command selection not supported");
+                        return "";
+                    }
+                    if (index === selectionEnd.line)
+                    {
+                        command = command.replace("`", "") + line.substring(0, selectionEnd.character);
+                    }
+                    else
+                    {
+                        command = command.replace("`", "") + line;
+                    }
                 }
-            }, 50);
+                return command;
+            }
         }
-        this.commandRunningStatusBarItem.show();
-        clearTimeout(this.hideStatusBarItemTimeout);
-
-        this.parsedResult = undefined;
-        this.query = undefined; // TODO
-        const cursor = source.selection.active;
-        const line = source.document.lineAt(cursor).text;
-        return this.findResultDocument()
-            .then(document => window.showTextDocument(document, ViewColumn.Two, true))
-            .then(target => replaceContent(target, JSON.stringify({ 'Running command': line }) + '\n')
-                .then(() => exec(line))
-                .then(({ stdout }) => stdout, ({ stdout, stderr }) => JSON.stringify({ stderr, stdout }, null, '    '))
-                .then(content => replaceContent(target, content)
-                        .then(() => this.parsedResult = JSON.parse(content))
-                        .then(undefined, err => {})
-                )
-                .then(() => this.commandFinished(t0))
-            )
-            .then(undefined, console.error);
+    }
+
+    private StripComments(text: string)
+    {
+        // allow for single line comments on the same line as the command (// or #)
+        var i = text.search("//");
```

Comment:
`//` is not supported for comments. Let's not add it here.

Reply:
This would only be in an azcli scrapbook; what would you think about making this a config setting? (only pushing back because I have a lot of C# muscle memory :) )

Reply:
It should still be possible to copy code between an azcli scrapbook and a normal shell script. Allowing `//` for comments works against that. Not convinced this warrants a setting. 🤔

Reply:
Hadn't considered that. Fair point - I'll take out the //.